src/types.ts

```diff
@@ -535,6 +535,7 @@ export type ZodStringCheck =
       kind: "datetime";
       offset: boolean;
       precision: number | null;
+      unqualified: boolean;
```

Comment:
Some other names I considered for this option, open to changing:
- `local`
- `utc` (defaults to true, set to false to allow unqualified dates)